### ISSUE TITLE
Community Assistance: Encrypted Client Hello

### ISSUE THREAD

**Issue by nhorman** [openssl/project#892]
There is community work ongoing to implement Encrypted Client Hello.  This epic tracks our efforts to ensure that work is reviewed and implemented 

Issues/PRs for this effort are tracked on the "Encrypted Client Hello" Milestone.  As a community effort we can not yet target a specific release, but during the course of our release development, we should ensure that any work done there is reviewed in a timely manner
```[tasklist]
### Tasks
- [ ] https://github.com/openssl/openssl/pull/24738
- [ ] Review 2 for https://github.com/openssl/openssl/pull/25193
```

**Comment by sftcd** [comment]
Hiya, for scheduling purposes, it'd be great to know when this is likely to start up. I assume @nhorman the kick-off is you create the feature branch, then tell me what you'd like as a 1st PR for that, I go make that PR and then we iterate? FWIW, I'm happy to start on that anytime, but, for me,  the more we can get done over the summer (before teaching restarts), the better.

**Comment by nhorman** [comment]
So we're still in the planning stages here, we're not even sure if this is something we can commit to yet, though we really appreciate your willingness to work on this.  The current plan is:
1) Determine what the core developer team commitment to this might be
2) Document that and get approval from openssl leadership
3) Mark it as planned/todo or rejected on our planning board

That all said, you are correct in that the technical process is going be the creation of a feature branch, for which you can open an issue to do so, which will then get considered and acted upon, at which time you would be free to iterate on it, and that can be done independently of any planning we have to do here.

**Comment by mattcaswell** [comment]
To create the feature branch there is a process:

https://www.openssl.org/policies/general/feature-branch-approval-policy.html

> Submitting a request for a new feature branch should be done by creating an issue in the [openssl/project](https://github.com/openssl/project) repository. Such a request can be created by anyone. The request should specify:
> 
> - The name of the feature branch requested in the form of feature/<name>
> - A link to an issue describing the requested feature that will be developed on the branch
> - An estimate of when the work on the feature is expected to be complete
> - Whether there are any known resources being made available to work on the feature

Once approved creating the branch is a simple matter.

@sftcd - I'd suggest you are best place to create that issue

**Comment by sftcd** [comment]
Thanks both. Will do that feature branch request later today.

**Comment by sftcd** [comment]
Thanks to whomever decided for creating the feature branch! I made an initial PR for that that just adds the doc/designs/ech-api.md file. I'm sure someone will tell me if I did the wrong github things, which is always possible;-) That's https://github.com/openssl/openssl/pull/24738.

Also - is the plan to keep this issue open as a place to sync up and e.g. discuss what next to do with the feature branch etc.?

**Comment by nhorman** [comment]
Yes, I'm not aware of any other issue open for ECH, so we can track the effort here

**Comment by nhorman** [comment]
@vdukhovni request that you co-ordinate with @sftcd to identify remaining prs that need review here so we can build out a set of review tasks

**Comment by pixitha** [comment]
Just a heads up, this is ratified now: https://www.rfc-editor.org/rfc/rfc9849.html

**Comment by mattcaswell** [comment]
This issue can be closed now. ECH is in the master branch and will be in 4.0 (to be released in April).
